docker-compose.yml

```diff
@@ -24,6 +24,16 @@ x-quorum-def:
     - /bin/sh
     - -c
     - |
+      UDS_WAIT=10
+      for i in $$(seq 1 100)
+      do
+        if [ -S $${PRIVATE_CONFIG} ];
```

Comment:
the socket file exists might not necessarily mean that the tx manager is fully up and running. There is an api /upcheck which can be called via http or unix-socket to verify this - you can find it was used in tessera-start.sh script.

Reply:
@namtruong I also created a suggestion for improvement for the `tessera-start.sh` script:

https://github.com/jpmorganchase/quorum-examples/issues/146

Reply:
Yes the `tessera-start.sh` script use curl via http instead of unix-socket.
The support for `curl` could easily be added - you can try something like `apk add curl`

Reply:
That's a bit of complication and then I will need another PR to `quorum` first to update Docker files:

* https://github.com/jpmorganchase/quorum/blob/8c339adc6c69e1ab408f3d3fa21a81d1715eb96e/Dockerfile
* https://github.com/jpmorganchase/quorum/blob/4b44bea1d7f38382d22f810e598f5f15aba5856e/Dockerfile.alltools

The docker-compose script here should not run `apk add curl` even if it needs `curl`.

Reply:
Yes I checked `wget` and it does not work with UDS, hence `curl` :)

Reply:
don't need to do `/upcheck` over UDS. Tessera opens HTTP `900x` and warrantee if `/upcheck` via HTTP is up, the UDS is ready to use. (line 123, 125)
https://github.com/jpmorganchase/quorum-examples/blob/633ed881e371f3c3bf4a15e77507fbd69de5c0f9/examples/7nodes/tessera-start.sh#L115-L128

Reply:
noted that need to use IP instead of localhost as Tessera doesn't bind on localhost

Reply:
`tessera-start.sh` is not for the Docker case, it's for running locally without Docker :) For Docker, Tessera & Quorum Geth actually communicate via UDS

Reply:
My suggestion is to update your logic to include `/upcheck` using HTTP (so to utilize existing `wget`) instead of UDS which requires `curl` or `nc`. This is similar to what `tessera-start.sh` does at the moment.

Reply:
Thanks @trung & @namtruong I used `wget` now to query the corresponding TxMgr to the Quorum Geth in the pair. Works great and we won't need `curl`.